### PR TITLE
fix(quick-chat): restore persisted tabs after task reload

### PR DESCRIPTION
**Pull request by carlosflorencio**
Task-detail reloads could hydrate Quick Chat from a stale active-workspace cookie before route hydration, leaving persisted chats in the backend while the dialog showed a blank workspace. Route-authoritative boot hydration and workspace-scoped tab selection now restore non-expired tabs in creation order without cross-workspace leakage.

## Important Changes

- Resolve the Quick Chat boot workspace from the current task route before cookie or user-setting fallbacks.
- Scope modal tabs, active-session workspace selection, blank setup tabs, and close fallbacks to one workspace.
- Cover two-tab reload restoration, creation order, persisted backend rows, and restored chat history on desktop and mobile.
- Update the existing Vitest mock signature for the current API so the aggregate typecheck remains green.

## Validation

- Reproduced on the production-build task-detail route with cookie workspace A and task workspace B; both workspace B Quick Chat rows remained persisted before and after reload.
- `GOCACHE=/tmp/kandev-go-cache make fmt`
- `GOCACHE=/tmp/kandev-go-cache make typecheck`
- `GOCACHE=/tmp/kandev-go-cache make test` (backend, 585 web files / 4721 tests, 30 CLI files / 278 tests, and script suites passed)
- `GOCACHE=/tmp/kandev-go-cache make lint`
- Production web build passed.
- Production-build desktop Quick Chat Playwright coverage passed: 7 tests, including two-tab reload restoration and selecting both restored histories.
- Production-build mobile Quick Chat Playwright coverage passed: 1 test, including two-tab reload restoration, creation order, and no overflow.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/1703?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/backend/internal/backendapp/boot_state.go [hunk 1]
```diff
@@ -310,6 +323,28 @@ func (b bootStateBuilder) resolveQuickChatWorkspaceID(ctx context.Context, req *
 	)
 }
 
+func (b bootStateBuilder) quickChatTaskRouteWorkspaceID(
+	ctx context.Context,
+	route webapp.RouteClassification,
+) string {
+	if route.Route != webapp.RouteTaskDetail || b.p.taskSvc == nil {
+		return ""
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/backendapp/boot_state.go]: Fixed in 4dcbd8a. Office task-detail routes now expose taskId in the route classification, Quick Chat accepts RouteOffice task metadata, and the boot regression covers both /t/:id and /office/tasks/:id with a stale workspace cookie.

File: docs/specs/tasks/quick-chat-expiration.md [hunk 2]
```diff
@@ -78,20 +85,22 @@ robust to whichever row a message/turn bumps. A quick chat expires when
...
-| **Idle longer than retention window** | **background sweep** | **new** |
+| Trigger                                     | Actor                | Existing?                                            |
+| ------------------------------------------- | -------------------- | ---------------------------------------------------- |
+| User closes a tab (confirm)                 | user                 | yes (`use-quick-chat-modal.ts` `handleConfirmClose`) |
+| Rapid re-pick supersedes an in-flight start | frontend             | yes (`useAgentSelection` orphan cleanup)             |
+| Agent profile deleted                       | user                 | yes (`DeleteEphemeralTasksByAgentProfile`)           |
+| **Idle longer than retention window**       | **background sweep** | **new**                                              |
 
 All four converge on `Service.DeleteTask`, so directory cleanup and event
 publishing are identical.
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply docs/specs/tasks/quick-chat-expiration.md]: Fixed in b87848a. The spec now records that expiration rechecks the predicate at delete time before using the shared cleanup and task.deleted event path.

File: apps/backend/internal/backendapp/boot_state.go [hunk 3]
```diff
@@ -284,7 +289,15 @@ func (b bootStateBuilder) addQuickChatState(ctx context.Context, req *http.Reque
...
 }
 
-func (b bootStateBuilder) resolveQuickChatWorkspaceID(ctx context.Context, req *http.Request, state map[string]any) string {
+func (b bootStateBuilder) resolveQuickChatWorkspaceID(
+	ctx context.Context,
+	req *http.Request,
+	state map[string]any,
+	route webapp.RouteClassification,
+) string {
+	if workspaceID := b.quickChatTaskRouteWorkspaceID(ctx, route); workspaceID != "" {
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/backendapp/boot_state.go:298]: No change. The added GetTask call is a single indexed task-row lookup; it does not load repository associations. Reusing taskDetailRouteData would couple initial Quick Chat hydration to the richer route-data DTO and still would not cover RouteOffice, where route data is intentionally absent. Keeping the route workspace lookup isolated preserves both boot paths with minimal cost.

File: apps/backend/internal/webapp/routes.go [hunk 4]
```diff
@@ -94,14 +94,22 @@ func classifySPARoute(requestPath string) (RouteName, map[string]string) {
...
 	}
 }
 
+func officeRouteParams(requestPath string) map[string]string {
+	taskID, ok := cutSingleSegment(requestPath, "/office/tasks/")
+	if !ok {
+		return nil
+	}
+	return map[string]string{"taskId": taskID}
+}
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/webapp/routes.go:111]: The current SPA defines only /office/tasks/:id. Advanced mode is a query parameter, which normalization strips before classification, and there are no nested task-detail routes. Rejecting additional path segments is deliberate so unrelated or future office paths are not misclassified as a task detail. The classifier can be extended alongside any future nested route.

File: apps/backend/internal/backendapp/boot_state.go [hunk 5]
```diff
@@ -310,6 +323,28 @@ func (b bootStateBuilder) resolveQuickChatWorkspaceID(ctx context.Context, req *
...
+	task, err := b.p.taskSvc.GetTask(ctx, taskID)
+	if err != nil {
+		b.logBootError("get quick-chat task route workspace", err)
+		return ""
+	}
+	if task == nil {
+		return ""
+	}
+	return task.WorkspaceID
+}
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/backendapp/boot_state.go:346]: Thanks. The lookup is intentional because the route task is the authoritative workspace source. addOfficeRouteState hydrates active-workspace aggregates but does not fetch the target office task. The /t route data fetch happens in a separate boot-data pass after initial state, so sharing it would require coupling quick-chat hydration to the route-data shape for one indexed primary-key read. Keeping the lookup explicit preserves the hydration boundary.

File: apps/web/components/quick-chat/quick-chat-provider.tsx [hunk 6]
```diff
@@ -14,18 +14,17 @@ function useIsMounted() {
...
-  if (isOpen) {
-    return activeWorkspace;
-  }
-  return null;
+  if (!isOpen) return null;
+  return (
+    sessions.find((session) => session.sessionId === activeSessionId)?.workspaceId ??
+    activeWorkspace
+  );
 }
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/components/quick-chat/quick-chat-provider.tsx:28]: This is intentional. useSettingsData(true) remains in QuickChatProvider, so agent profiles still preload globally; only modal internals wait until the dialog opens. That avoids pre-mounting the modal against the first restored session from a different workspace while preserving first-open setup data.

File: docs/specs/tasks/quick-chat-expiration.md [hunk 7]
```diff
@@ -1,12 +1,13 @@
 ---
-status: draft
+status: shipped
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply docs/specs/tasks/quick-chat-expiration.md:2]: Updated the shipped spec to record the verified behavior: raw message/turn writes only touch their own rows, while the prompt lifecycle updates the session timestamp and may update the task timestamp. The expiration query reads both and excludes active RUNNING/IDLE sessions.
